benches/benches/db_lookup_times_utils/seed.rs

```diff
@@ -0,0 +1,153 @@
+        seed_full_blocks(&mut database, block_count, tx_count);
+    }
+}
+
+fn generate_bench_block(height: u32, tx_count: u32) -> Block {
+    let header = PartialBlockHeader {
+        application: Default::default(),
+        consensus: ConsensusHeader::<Empty> {
+            height: BlockHeight::from(height),
+            ..Default::default()
+        },
+    };
+    let txes = generate_bench_transactions(tx_count);
+    let block = PartialFuelBlock::new(header, txes);
+    block.generate(&[], Default::default()).unwrap()
+}
+
+fn generate_bench_transactions(tx_count: u32) -> Vec<Transaction> {
+    let mut txes = vec![];
+    for _ in 0..tx_count {
+        txes.push(Transaction::default_test_tx());
+    }
+    txes
+}
+
+fn height_key(block_height: u32) -> Vec<u8> {
+    BlockHeight::from(block_height).to_bytes().to_vec()
+}
+
+fn insert_compressed_block(
+    database: &mut RocksDb<OnChain>,
+    height: u32,
+    tx_count: u32,
+) -> Block {
+    let block = generate_bench_block(height, tx_count);
+
+    let compressed_block = block.compress(&chain_id());
+    let height_key = height_key(height);
+
+    let raw_compressed_block = postcard::to_allocvec(&compressed_block).unwrap().to_vec();
+    let raw_transactions = block
+        .transactions()
+        .iter()
+        .map(|tx| {
+            (
+                tx.id(&chain_id()),
+                postcard::to_allocvec(tx).unwrap().to_vec(),
+            )
+        })
+        .collect::<Vec<_>>();
+
+    // 1. insert into CompressedBlocks table
+    database
+        .put(
+            height_key.as_slice(),
+            Column::FuelBlocks,
+            Value::new(raw_compressed_block),
+        )
+        .unwrap();
+    // 2. insert into Transactions table
+    for (tx_id, tx) in raw_transactions {
+        database
+            .put(tx_id.as_slice(), Column::Transactions, Value::new(tx))
+            .unwrap();
+    }
+
+    block
+}
+
+fn insert_full_block(database: &mut RocksDb<OnChain>, height: u32, tx_count: u32) {
+    let block = insert_compressed_block(database, height, tx_count);
+
+    let height_key = height_key(height);
+    let raw_full_block = postcard::to_allocvec(&block).unwrap().to_vec();
+
+    // 3. insert into FullFuelBlocks table
+    database
+        .put(
+            height_key.as_slice(),
+            Column::FullFuelBlocks,
+            Value::new(raw_full_block),
+        )
+        .unwrap();
+}
+
+fn seed_compressed_blocks_and_transactions(
+    database: &mut RocksDb<OnChain>,
+    block_count: u32,
+    tx_count: u32,
+) -> Vec<Block> {
+    let mut blocks = vec![];
+    for block_number in 0..block_count {
+        blocks.push(insert_compressed_block(database, block_number, tx_count));
+    }
+    blocks
+}
+
+fn seed_full_blocks(database: &mut RocksDb<OnChain>, block_count: u32, tx_count: u32) {
+    // we seed compressed blocks and transactions to not affect individual
+    // lookup times
```

Comment:
This comment doesn't make sense to me inside the context of `seed_full_blocks`. I'm also not sure what it means to "affect individual lookup times".

Reply:
the context is that if we do include `FullFuelBlocks` in the codebase at some point and use it, we want to use it in tandem with the `FuelBlocks` and `Transactions` tables so as to not affect their individual lookup times, i.e cases where we make individual block header calls (to get the block height, for example), and where we make individual tx lookups.

Reply:
I understand now. I think this comment should be above the `insert_full_block` in that case.

Reply:
addressed in 7cca27818